managed/utils/encryption/encryption.go

```diff
@@ -106,8 +106,11 @@ func RotateEncryptionKey() error {
 	}
 
 	defaultEncryptionMtx.Lock()
-	DefaultEncryption = New()
-	defaultEncryptionMtx.Unlock()
+	defer defaultEncryptionMtx.Unlock()
+	DefaultEncryption, err = New()
+	if err != nil {
```

Comment:
Would 'return err' suffice?

Reply:
do you think we need to panic? it will be returned by stack to main